### DISCUSSION TITLE
How to type a custom color

### DISCUSSION THREAD

**Discussion by ShalltearFR** [danpacho/tailwindest#87] [Q&A] [answered]
Hello, I would like to know how to type custom colors in Tailwindest.
Indeed, I have this code:
```
interface Colors {
  title: Tailwindest["color"];
  desc: Tailwindest["color"];
}
```

tailwind.config.ts :
```
import type { Config } from "tailwindcss";

const config: Config = {
  content: [
    "./pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./components/**/*.{js,ts,jsx,tsx,mdx}",
    "./app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  theme: {
    extend: {
      colors: {
        primary: "white",
        secondary: "black",
        tertiary: "#667479",
        accent: "#ef7f01", //orange
      },
    },
  },
  plugins: [],
};
export default config;
```

The problem is that Tailwindest["color"] does not take into account the custom colors contained in tailwind.config.ts.
The execution works, the error is only typescript.

Type error :
`Type '"text-accent"' is not assignable to type '"text-slate-100" | "text-slate-200" | "text-slate-300" | "text-slate-500" | "text-slate-700" | "text-slate-50" | "text-slate-400" | "text-slate-600" | "text-slate-800" | ... 250 more ... | undefined'.ts(2322)`

tailwind version 3.3.6
Tailwindest version 2.2.1

**Comment by danpacho** [comment, marked as answer]
Oh you can add custom properties with `Tailwindest` type's generic argument like this. Happy coding :)

```ts
export type Tw = Tailwindest<{
	color: "primary" | "secondary" | "tertiary" | "accent"
}>

// access background color
type Bg = Tw["backgroundColor"]
```

**Reply by ShalltearFR** [reply]
hooooo thanks
